js/modules/k6/http/http_request.go

```diff
@@ -334,7 +356,9 @@ func (h *HTTP) request(ctx context.Context, rt *goja.Runtime, state *common.Stat
 
 	if resErr != nil {
 		resp.Error = resErr.Error()
-		tags["error"] = resp.Error
+		if state.Options.DefaultTags == nil || state.Options.DefaultTags["error"] {
+			tags["error"] = resp.Error
+		}
```

Comment:
Shouldn't we actually update the `status` and `proto` fields here, even though there was an error. Arguably the `status` field is even more important in this case

Reply:
Agree, we should aim to set the tag values in connection with the context that the value reflects. So for `status` we should set it to 0 when handling an error, and `proto` should be set to an empty string.

Reply:
Ah, I mistakenly assumed that non-nil `resErr` meant an HTTP error here, when it's mostly for things like network connectivity errors and such. So I'm not actually sure we should set `status` to 0 and `proto` to an empty string. Wouldn't it be better if we just don't include them at all, since 0 and an empty string are nonsensical values?

Reply:
Yes, it's (mostly) for non/pre-HTTP errors. `status=0` is used in the cloud service in various places when querying the result data. We might come up with a better solution eventually, perhaps by only looking at the `error` tag, but not high priority right now. @ppcano Can answer better than me how reliant we currently are on `status=0` in the cloud service.

Reply:
Sorry, my latest push hid this comment section, hopefully commenting here will restore it, since my question is still valid with the new code. @ppcano, if this section still shows as outdated, please respond in the main PR comment thread.